### PR TITLE
Will/20230612 voice input

### PR DESCRIPTION
**Pull request by williamtriinh**
Add speech to text using Azure Cognitive Services

Screenshot 2023-06-18 at 19 52 47

### CODE REVIEW

File: utils/app/i18n.ts [hunk 1]
```diff
@@ -24,4 +24,31 @@ function getAvailableLocales() {
   ];
 }
 
-export { getAvailableLocales };
+// https://learn.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support?tabs=stt#supported-languages
+function getAvailableSpeechSourceLanguages() {
+  return [
+    { name: 'English', value: 'en-US'},
+    { name: '官話', value: 'zh-CN' },
+    { name: '廣東話', value: 'zh-HK' },
```
Comment (williamtriinh) [inline utils/app/i18n.ts]: @thejackwu Could you confirm that line 31 is Mandarin and the one below is Cantonese lol.

Comment (thejackwu) [reply utils/app/i18n.ts]: lol Yep line 31 should be "國語"

File: utils/app/outputLanguage copy.ts
Comment (thejackwu) [inline utils/app/outputLanguage copy.ts:1]: I assume this file shouldn't be here?

**Comment by thejackwu** [comment]
https://github.com/exploratortech/chat-everywhere/assets/4432048/6d051e82-f2a9-4c7c-982a-c4ea3e2954db

It's awesome!!!!! But found a potential bug, if I click send right while recording, the overlay will still persist.

**Comment by 1orZero** [comment]
Screenshot 2023-06-19 at 9 21 35 AM

It looks like the regenerate button is currently being covered.

As most of our users are from Taiwan, may I assume that the output in Mandarin should be in traditional Chinese characters?

**Comment by 1orZero** [comment]
@thejackwu  @williamtriinh  I have added the option Mandarin (Traditional Chinese) option.

Screenshot 2023-06-19 at 9 52 41 AM

File: utils/app/outputLanguage copy.ts
Comment (williamtriinh) [reply utils/app/outputLanguage copy.ts:1]: Ah it's suppose to be but, I forgot to change the file name

**Comment by williamtriinh** [comment]
@1orZero fixed the regenerate button getting covered. @thejackwu I actually had the intention of not allowing the user to send prompts while the voice recognition was active to avoid any weird behaviour.

**Review by thejackwu — APPROVED** [review]

**Pull request merged**